omnibus/partybus/lib/partybus/migration_api/v1.rb

```diff
@@ -97,19 +97,40 @@ def run_command(command, options={})
         runner.run_command(command, options)
       end
 
-      def run_sqitch(ec_target, osc_target="@1.0.4")
-        args =  " --db-user #{Partybus.config.postgres['db_superuser']}"
-        args << " --db-name opscode_chef"
-        args << " --db-host #{Partybus.config.postgres['vip']}"
-        args << " --db-port #{Partybus.config.postgres['port']}"
-        args << " --engine pg"
-
-        cmd = "cd /opt/opscode/embedded/service/opscode-erchef/schema"
-        cmd << " && cd baseline && sqitch #{args} deploy --to-target #{osc_target} --verify"
-        cmd << " && cd .. && sqitch #{args} deploy --to-target #{ec_target} --verify;"
-        run_command(cmd,
-                    :cwd => "/opt/opscode/embedded/service/opscode-erchef/schema",
-                    :env => {"PGPASSWORD" => Partybus.config.postgres['db_superuser_password']})
+      # Caller may use 'opts' to override :username :password,
+      # :database, and/or :path.
+      def run_sqitch(target, service, opts = {})
+        options = default_opts_for_service(service).merge(opts)
+        command = <<-EOM.gsub(/\s+/," ").strip!
+          sqitch --engine pg
+            --db-name #{options[:database]}
+            --db-host #{Partybus.config.postgres['vip']}
+            --db-port #{Partybus.config.postgres['port']}
+            --db-user #{options[:username]}
+            --top-dir /opt/opscode/embedded/service/#{options[:path]}
+            deploy #{target} --verify
+        EOM
+        run_command(command, env: {"PGPASSWORD" => options[:password]})
+      end
+
+      def default_opts_for_service(service)
+        username = Partybus.config.secrets[service]['sql_user']
```

Comment:
The `sql_user` isn't stored in the `secrets` hash, which is why this is failing tests.